src/ai/susi/server/api/aaa/CheckRegistrationService.java

```diff
@@ -0,0 +1,89 @@
+/**
+ *  SignUpServlet
+ *  Copyright 16.06.2018 by Arundhati Gupta, @arundhati24
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public License
+ *  along with this program in the file lgpl21.txt
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package ai.susi.server.api.aaa;
+
+import ai.susi.DAO;
+import ai.susi.EmailHandler;
+import ai.susi.json.JsonObjectWithDefault;
+import ai.susi.server.*;
+import ai.susi.tools.IO;
+import ai.susi.tools.TimeoutMatcher;
+import org.json.JSONObject;
+
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.regex.Pattern;
+
+/**
+ * This endpoint accepts 1 parameter email.
```

Comment:
Change email to check_email

Reply:
The account identification is not necessarily email only. The account ID has generally an email prefix but it can be differently. Keep a generic approach

Reply:
@Orbiter As we use email address and password as the login credentials, there was a need to check if an email address has signed up for SUSI or not. This is needed to improve the auth flow and to make it similar to Eventbrite. The aim is to ask the user to enter an email address and then check whether that email address has already been signed up for SUSI or not. If that email address has already signed up, we would simply show the password field otherwise we would display the sign up screen with the email address already filled in. As per my understanding, we do not have any identification for an account unless it has signed up for SUSI, therefore, I kept only one parameter i.e. 'email address'.
It would be really very helpful if you could please reflect a bit upon what other parameters should be taken as per the requirement described above and also if there is something that I have missed. Thanks.